### PR TITLE
Enemies

### PR DESCRIPTION
**Pull request by ghost**
Patrol and Chase implemented in prototype scene
You can ignore everything in Assets/Materials/Thirdparty since it's just imported materials from https://assetstore.unity.com/packages/2d/textures-materials/gridbox-prototype-materials-129127
image

### CODE REVIEW

**Comment by kantagara** [comment]
One question: In regards to ExitStates... I would maybe suggest that there also exists some sort of transition. like if you're in patroling state, it would make sense you can go to smash state if player pops up right in front of you (So distance to target is something small), or to go to chase state if he's further away.

**Comment by kantagara** [comment]
Other than that, looks solid, will approve.

**Review by kantagara — APPROVED** [review]

**Review by sneakzttv — APPROVED** [review]

**Review by kantagara — APPROVED** [review]

File: Assets/Scripts/NPC/Transitions/PlayerOutOfSightTransition.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Chainsafe.BlockBlasterz
+{
+    public class PlayerOutOfSightTransition : Transition<ChaseState>
+    {
+        public override bool TryExitState()
+        {
+            if (FromState.OutOfSightTimeout <= 0)
```
Comment (kantagara) [inline Assets/Scripts/NPC/Transitions/PlayerOutOfSightTransition.cs:11]: This could be shorter? Just return FromState.OutOfSightTimeout <= 0

Comment (rob1997) [reply Assets/Scripts/NPC/Transitions/PlayerOutOfSightTransition.cs]: yep, changes made

**Pull request merged**